### PR TITLE
Update spackage in preparation for release

### PR DESCRIPTION
**Pull request by jhp-lanl**
## PR Summary

We have updated some of our dependencies since the last version. This change simply updates the `package.py` to reflect that the new version will depend on these new packages.

Specifically, the two main updates are
- ports-of-call to 1.6.0
- Catch2 to version 3

## PR Checklist

- N/A Adds a test for any bugs fixed. Adds tests for new features.
- N/A Format your changes by using the `make format` command after configuring with `cmake`.
- N/A Document any new features, update documentation for changes made.
- [x] Make sure the copyright notice on any files you modified is up to date.
- N/A After creating a pull request, note it in the CHANGELOG.md file.
- [x] LANL employees: make sure tests pass both on the github CI and on the Darwin CI

If preparing for a new release, in addition please check the following:
- [ ] Update the version in cmake.
- [ ] Move the changes in the CHANGELOG.md file under a new header for the new release, and reset the categories.

### CODE REVIEW

**Comment by jhp-lanl** [comment]
@rbberger can you review this?

Can you also take a glance at the `package.py` and make sure I didn't miss anything that needs to be updated for the new version?

File: spack-repo/packages/singularity-eos/package.py [hunk 1]
```diff
@@ -96,8 +96,9 @@ class SingularityEos(CMakePackage, CudaPackage):
 
     # building/testing/docs
     depends_on("cmake@3.19:")
-    depends_on("catch2@3.0.1:", when="@main +tests")
     depends_on("catch2@2.13.7", when="@:1.8.0 +tests")
+    depends_on("catch2@3.0.1:", when="@1.9.0: +tests")
+    depends_on("catch2@3.0.1:", when="@main +tests")
```
Comment (rbberger) [inline spack-repo/packages/singularity-eos/package.py:101]: The `@main` line is not needed. `1.9.0:` is "1.9.0 and anything newer, which `main`, `master` and `develop` always are.

**Comment by rbberger** [comment]
@jhp-lanl I think all the plugins logic should be changed from `@main` to `@1.9.0:`.

File: spack-repo/packages/singularity-eos/package.py (hunk 1 above)
Comment (jhp-lanl) [reply spack-repo/packages/singularity-eos/package.py]: Done

**Comment by jhp-lanl** [comment]
> @jhp-lanl I think all the plugins logic should be changed from `@main` to `@1.9.0:`.

I believe this should be good now.

@Yurlungur I think this MR is ready

**Review by Yurlungur — APPROVED** [review]

**Pull request merged**